libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java

```diff
@@ -121,6 +123,7 @@ public NavigationView(Context context, @Nullable AttributeSet attrs, int defStyl
     ThemeSwitcher.setTheme(context, attrs);
     initializeView();
     lifecycleRegistry = new LifecycleRegistry(this);
+    onNavigationReadyCallbacks.add(internalNavigationReadyCallback);
```

Comment:
where are we removing it from the `Set`?

Reply:
Why would we remove it from the set? The NavigationView owns the set and the callback. When the NavigationView gets destroyed everything goes with it.